src/renderer/src/lib/Music.ts

```diff
@@ -221,42 +191,38 @@ createEffect(async () => {
   setLocalVolume(audio.value.volume ?? 0.5);
 });
 
-
-
 createEffect(() => {
   player.volume = calculateVolume();
 });
 
-
-
-const [writeVolume, ] = delay(async (volume: number) => {
+const [writeVolume] = delay(async (volume: number) => {
   await window.api.request("settings::write", "volume", volume);
 }, 200);
 createEffect(async () => {
   const v = volume();
   writeVolume(v);
 });
 
-
-
 createEffect(async () => {
   const lv = localVolume();
 
-  if (isSongUndefined(song) || lv === 0.5) {
+  if (isSongUndefined(song()) || lv === 0.5) {
```

Comment:
Moved the logic that saves the local volume from the effect into the local volume setter function, this will prevent weird side effects (like the volume being saved when the song is updated).